crates/red_knot/docs/tracing.md

```diff
@@ -0,0 +1,103 @@
+# Tracing
+
+Traces are a useful tool to narrow down the location of a bug or, at least, to understand why the compiler is doing a particular thing.
+Note, tracing messages with severity `debug`..`error` are user-facing. They should be phrased accordingly.
```

Comment:
Very minor nit: Doesn't "`debug`..`error`" technically exclude `error`, if we assume Rust range syntax, which is what this seems to be emulating? But `error` is kind of the most user-facing, so probably we shouldn't be ambiguous about this. Maybe just say something like "`debug` or greater severity"?